### ISSUE TITLE
why cli results is different with python

### ISSUE THREAD

**Issue by ankhafizov** [ultralytics/ultralytics#18892]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

![Image](https://github.com/user-attachments/assets/0c2516ae-099b-4930-a2fa-d9717df14eca)

vs

```
yolo predict model=yolo11m.pt source=video.avi show
```

python code gives no detections. Why? Weights are the same (default)?

CLI results:

video 1/1 (frame 11/54025) C:\repos\restaurant\detection\video.avi: 480x640 5 persons, 19 chairs, 4 potted plants, 6 dining tables, 1 laptop, 1 vase, 11.7ms

**Comment by UltralyticsAssistant** [comment]
👋 Hello @ankhafizov, thank you for your interest in Ultralytics 🚀! We recommend checking out the [Docs](https://docs.ultralytics.com/) for additional guidance, including detailed references to both [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples. Many common questions may already be answered there.

If this is a 🐛 Bug Report, please help us debug it by providing a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/), including:
- The exact Python script you're using
- The CLI command producing different results
- Any console logs or error messages

If this is a ❓ Question about your results, please include as much detailed information as possible to help us assist you effectively, such as:
- Model details (e.g., `yolo11m.pt`)
- Inputs (e.g., `video.avi`)
- Any relevant configuration details or deviations from the default setup

In the meantime, make sure your environment is up to date to rule out version-related inconsistencies. You can upgrade the `ultralytics` package and dependencies in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

For further experimentation or testing, YOLO can be run in the following verified environments:
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

Join the Ultralytics community to get insights from other users! Whether it’s real-time help on [Discord](https://discord.com/invite/ultralytics) 🎧, engaging threads on [Discourse](https://community.ultralytics.com/), or a discussion on our [Subreddit](https://reddit.com/r/Ultralytics), there's a place for everyone.

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests ensure all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) are functioning correctly across macOS, Windows, and Ubuntu environments.

This is an automated response to help guide you 😊. An Ultralytics engineer will also take a closer look at your issue soon.

**Comment by Y-T-G** [comment]
`yolo11m.pt` is the pretrained model.

`yolo11m.yaml` is a randomly initialized model which is not trained.
